staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -975,3 +985,23 @@ func (az *Cloud) ShouldNodeExcludedFromLoadBalancer(nodeName string) (bool, erro
 
 	return az.excludeLoadBalancerNodes.Has(nodeName), nil
 }
+
+
+func isNodeReady(node *v1.Node) bool {
+
+	if len(node.Status.Conditions) == 0 {
+		klog.V(4).Infof("considering node %v with no Status.Conditions as not ready",
+			node.ObjectMeta.Name)
+		return false
+        }
+
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady && cond.Status != v1.ConditionTrue {
```

Comment:
This should be inverted.  If the condition is true, return true.  Then return false at the end of the function.